### PR TITLE
Move FastMemcpy to contribs

### PR DESCRIPTION
**Pull request by abyss7**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Non-significant (changelog entry is not required)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -121,12 +121,8 @@ endif ()
 
 option (ENABLE_TESTS "Enables tests" ON)
 
-if (ARCH_AMD64)
-    option (USE_INTERNAL_MEMCPY "Use internal implementation of 'memcpy' function instead of provided by libc. Only for x86_64." ON)
-endif ()
-
 if (OS_LINUX AND NOT UNBUNDLED AND MAKE_STATIC_LIBRARIES AND NOT SPLIT_SHARED_LIBRARIES AND CMAKE_VERSION VERSION_GREATER "3.9.0")
-    option (GLIBC_COMPATIBILITY "Set to TRUE to enable compatibility with older glibc libraries. Only for x86_64, Linux. Implies USE_INTERNAL_MEMCPY." ON)
+    option (GLIBC_COMPATIBILITY "Set to TRUE to enable compatibility with older glibc libraries. Only for x86_64, Linux" ON)
```
Comment (alexey-milovidov) [inline CMakeLists.txt]: But without our own memcpy GLIBC_COMPATIBILITY does not make sense because does not ensure compatibility.

Comment (alexey-milovidov) [reply CMakeLists.txt]: Ok, I see that it still implies usage of internal memcpy.

Comment (abyss7) [reply CMakeLists.txt]: Will return the comment back.

File: base/memcpy/README.md [hunk 2]
```diff
@@ -1,20 +0,0 @@
-Internal implementation of `memcpy` function.
```
Comment (alexey-milovidov) [inline base/memcpy/README.md]: This comment should not be removed.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Ok except the comment that must remain.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
```
../contrib/FastMemcpy/FastMemcpy.c:24:2: error: it can only be compiled under windows or unix
#error it can only be compiled under windows or unix
 ^
```

**Comment by alexey-milovidov** [comment]
```
../contrib/FastMemcpy/FastMemcpy.h:159:29: runtime error: load of misaligned address 0x00000769aadd for type 'uint64_t' (aka 'unsigned long'), which requires 8 byte alignment
0x00000769aadd: note: pointer points here
 45 52 45 20 6e 61 6d  65 20 4c 49 4b 45 20 00  52 65 63 65 69 76 65 64  20 70 61 63 6b 65 74 20  77
             ^ 
    #0 0x20ee1ced in memcpy_tiny /build/obj-x86_64-linux-gnu/../contrib/FastMemcpy/FastMemcpy.h:159:29
    #1 0x20ee1ced in memcpy_fast /build/obj-x86_64-linux-gnu/../contrib/FastMemcpy/FastMemcpy.h:683
    #2 0x15a46179 in std::__1::char_traits<char>::copy(char*, char const*, unsigned long) /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/__string:269:56
    #3 0x15a46179 in std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::__init(char const*, unsigned long) /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:1793
    #4 0x15a46179 in std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string<std::nullptr_t>(char const*) /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:819
    #5 0x15a46179 in __cxx_global_var_init.43 /build/obj-x86_64-linux-gnu/../dbms/programs/server/MySQLHandler.cpp:366
    #6 0x15a46179 in _GLOBAL__sub_I_MySQLHandler.cpp /build/obj-x86_64-linux-gnu/../dbms/programs/server/MySQLHandler.cpp
    #7 0x21da701c in __libc_csu_init (/usr/bin/clickhouse+0x21da701c)
    #8 0x7fc4b092bb27 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b27)
    #9 0x15898029 in _start (/usr/bin/clickhouse+0x15898029)
```

**Comment by alexey-milovidov** [comment]
It looks like FastMemcpy should be compiled without UBSan.

**Comment by alexey-milovidov** [comment]
You forgot to compare the MemCpy source code for patches.

**Pull request merged**